### PR TITLE
add config_clean.sh as a way to clean up MLZ resources

### PR DESCRIPTION
**Pull request by glennmusa**
# Description

Provide a mechanism to clean up all the resources created by Mission LZ to support a deployment.

## Issue reference

The issue this PR will close: #4, #150

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles or validates correctly
* [x] BASH scripts have been validated using `shellcheck`
* [x] All tests pass (manual and automated)
* [x] The documentation is updated to cover any new or changed features
* [x] Markdown files have been linted using the recommended linter. (See `.vscode/extensions.json`.)
* [x] Relevant issues are linked to this PR

### CODE REVIEW

**Review by brooke-hamilton — APPROVED** [review]
Everything worked as designed.

**Review by brooke-hamilton — CHANGES_REQUESTED** [review]
My mistake, it looks like the cleanup script is deleting all the resources, including the hub and T0. That is not the intended behavior, right?

```TEXT
INFO: deleting rg-mlz-tf-mlz1618577267...
INFO: deleting rg-tf-sacahub-mlz1618577267...
INFO: deleting rg-tf-tier0-mlz1618577267...
INFO: deleting rg-tf-tier1-mlz1618577267...
INFO: deleting rg-tf-tier2-mlz1618577267...
INFO: deleting rg-saca-mlz1618577267...
INFO: deleting rg-t0-mlz1618577267...
INFO: deleting rg-t1-mlz1618577267...
INFO: deleting rg-t2-mlz1618577267...
INFO: deleting service principal sp-mlz-tf-mlz1618577267...
Removing role assignments
INFO: purging key vault mlzkvmlz1618577267d8f20d...
INFO: Complete! Resources for mlz1618577267 deleted!
```

**Comment by glennmusa** [comment]
@brooke-hamilton you're right, let me update how MLZ Configuration resources are tagged.

**Comment by glennmusa** [comment]
> My mistake, it looks like the cleanup script is deleting all the resources, including the hub and T0. That is not the intended behavior, right?
> 
> ```
> INFO: deleting rg-mlz-tf-mlz1618577267...
> INFO: deleting rg-tf-sacahub-mlz1618577267...
> INFO: deleting rg-tf-tier0-mlz1618577267...
> INFO: deleting rg-tf-tier1-mlz1618577267...
> INFO: deleting rg-tf-tier2-mlz1618577267...
> INFO: deleting rg-saca-mlz1618577267...
> INFO: deleting rg-t0-mlz1618577267...
> INFO: deleting rg-t1-mlz1618577267...
> INFO: deleting rg-t2-mlz1618577267...
> INFO: deleting service principal sp-mlz-tf-mlz1618577267...
> Removing role assignments
> INFO: purging key vault mlzkvmlz1618577267d8f20d...
> INFO: Complete! Resources for mlz1618577267 deleted!
> ```

This should be fixed @brooke-hamilton

**Comment by brooke-hamilton** [comment]
When I deployed everything to the same sub, this worked exactly as described. 

This scenario did not work: 
1. Deploy with each tier and the config in a different subscription, like this:

```BASH
  src/deploy.sh -s {my_mlz_configuration_subscription_id} \
    -h {my_hub_network_subscription_id} \
    -0 {my_identity_network_subscription_id} \
    -1 {my_operations_network_subscription_id} \
    -2 {my_shared_services_network_subscription_id} \
    -z {my_mlz_environment_name}
```
1. Run `config_clean.sh` like this:

```BASH
src/scripts/config/config_clean.sh src/generated-configurations/jbh002.mlzconfig 
INFO: deleting mlz-config-jbh002...
Resource group 'mlz-config-jbh002' could not be found.
```

**Comment by glennmusa** [comment]
> ```shell
> src/deploy.sh -s {my_mlz_configuration_subscription_id} \
>     -h {my_hub_network_subscription_id} \
>     -0 {my_identity_network_subscription_id} \
>     -1 {my_operations_network_subscription_id} \
>     -2 {my_shared_services_network_subscription_id} \
>     -z {my_mlz_environment_name}
> ```

That makes sense. I'll update to use the correct subscription. Thanks!

File: src/scripts/config/config_clean.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  for rg in $rgs_to_delete;
+  do
+    echo "INFO: deleting ${rg}..."
+
+    az group delete \
+      --subscription "${sub}" \
+      --name "${rg}" \
+      --yes \
+      --only-show-errors \
+      --output none
```
Comment (glennmusa) [inline src/scripts/config/config_clean.sh:70]: Can you try again for me @brooke-hamilton? Was missing the `--subscription` argument here.

**Review by brooke-hamilton — APPROVED** [review]

**Pull request merged**